### ISSUE TITLE
LoRA training for Wan 2.1 I2V?

### ISSUE THREAD

**Issue by Fannovel16** [tdrussell/diffusion-pipe#112]
Lora training for t2v is nice and all but I think it's useless when we already have official i2v

**Comment by tdrussell** [comment]
I can try to implement this when I get time. But there are some considerations to be aware of.

1. I think you must train a i2v model only on the conditioning images + the rest of the video. So basically video training. Only 14b has i2v. Video training on a model at this size is not possible for most users because it uses a lot of VRAM.
2. If the i2v was created by finetuning from the t2v (this was true for Skyreels for example), it may be better to just train a lora on t2v then use it with i2v. I think this would work for most types of concepts, and it would allow you to do image-only training to teach new concepts, which is a common scenario.
3. If I implemented i2v, I don't think I have any suitable datasets to train that would let me test that it's really working. I could of course just make sure training runs, and release it with the caveat that it's completely untested. Any testing would also be slow, because 14b model + you have to do video training as mentioned.

**Comment by Fannovel16** [comment]
@tdrussell 
2. I compared the weight of 14B t2v and 14B i2v and you might be right. i2v has different patch_embedding shape, has it own img_emb, blocks.n.cross_attn.k_img and blocks.n.cross_attn.v_img but other layers have the same shape as in t2v
3. Any video dataset works. Just use the first frame as conditioning image

**Comment by qiwang1996** [comment]
I tried  to develop i2v lora-training today  but the logic of utils/dataset.py is so complex that i choose to give up .... 
Why i tried to do this is that i trained good loras of wan-t2v through this repo and the result is even better than that of diffsynth-studio.

**Comment by qiwang1996** [comment]
You are really a good open-source contributor of this community. Looking forward to your nice implementation !!! @tdrussell

**Comment by Kvento** [comment]
I am also looking forward to the opportunity to train the i2v model.

**Comment by sogoojoy** [comment]
> I tried to develop i2v lora-training today but the logic of utils/dataset.py is so complex that i have to give up .... Why i tried to do this is that i trained good loras of wan-t2v through this repo and the result is even better than that of diffsynth-studio.
Could you please share the config of train wan t2v lora? Thank you so much

**Comment by sogoojoy** [comment]
> I tried to develop i2v lora-training today but the logic of utils/dataset.py is so complex that i have to give up .... Why i tried to do this is that i trained good loras of wan-t2v through this repo and the result is even better than that of diffsynth-studio.

Could you please share the config of train wan t2v lora? Thank you so much

**Comment by tdrussell** [comment]
Wan i2v training is now supported. I think it works? I don't really have a good dataset to test. The i2v is 14b parameters and it requires training on videos, so you're gonna need a lot of VRAM, probably 48GB.

**Comment by qiwang1996** [comment]
how can you be so nice,  您真牛逼,   can you share a QR code in the project，i think there are people like me who want to pay for a cup of coffee for your good efforts.

**Comment by qiwang1996** [comment]
https://github.com/Tencent/HunyuanVideo/issues/219  

I created an issue about first frame encoding and i am curious about results of test that e1 != ee1.  Why can we use the whole video latents and just extract  first frame embedding of it, then padding zeros after that?

**Comment by qiwang1996** [comment]
https://github.com/modelscope/DiffSynth-Studio/blob/a66203a3914b6ee54240b3cdd1b03408c4f11ec7/diffsynth/pipelines/wan_video.py#L164

In their inference code, they also encode conditional embedding separately, which makes me curious.

**Comment by tdrussell** [comment]
Hmm, I am not sure about HunyuanVideo. In that linked issue I would also expect the first frame to be the same in both cases but I have never tried. For Wan I can say the following:
1. Official inference code creates the image conditioning by having the image as the first frame, and the rest of the frames as 0, then encoding that whole thing. ComfyUI does the same. Note that this is different from encoding just the first frame by itself and then padding that single latent frame with 0s, since the frames that were 0 for video do not end up as 0 for latent.
2. Consider the two scenarios: 1) encode whole video, extract the first latent frame, decode it. 2) Set all video frames except the first to 0, encode it, extract first latent frame, decode it. The decoded first frame is identical in both cases. I tested this myself in the wan_vae_test.py file I added.

**Comment by qiwang1996** [comment]
i reviewed your i2v training code and it aligns to inference. which is correct theoretically for wan👍 
There are actually  the mentioned small difference between Wan and Hunyuan about I2V conditional embedding as i read technical reports of them. 
Hunyuan:
![Image](https://github.com/user-attachments/assets/a79b8513-6b38-4b27-acc2-7f07be601163)
Wan:
![Image](https://github.com/user-attachments/assets/abdfc428-3bc5-473c-81f7-8732a41d5cb3)

**Comment by hellesgrind** [comment]
Hey there! Is it closed due to being resolved or you decided not to implement?

**Comment by OJLDev** [comment]
@tdrussell 
I tried to train a new lora model based on Wan 2.1 I2V 14B 720P
it is my data set
character1_face.mp4
character1_face.txt
character1_run.mp4
character1_run.txt
character1_stand.mp4
character1_stand.txt
character1_walk.mp4
character1_walk.txt
character2_face.mp4
character2_face.txt
character2_run.mp4
character2_run.txt
character2_stand.mp4
character2_stand.txt
character2_walk.mp4
character2_walk.txt

here, I aimed to keep learning character consistency of character1, character2.
But training process are not converged with correct trending.
Please help me to prepare correct dataset.
Also, regarding Wan 2.1 FLF2V 14B 720P, I need to create a new lora model for character consistency.
Please help me.
Regards
